packages/vite/src/node/plugins/css.ts

```diff
@@ -2759,8 +2759,9 @@ async function compileLightningCSS(
           ? config.css?.lightningcss?.cssModules ?? true
           : undefined,
       })
-
-  let css = res.code.toString()
+  // Deno res.code returns incorrect format and needs conversion to buffer
+  const code = res.code instanceof Uint8Array ? Buffer.from(res.code) : res.code
+  let css = code.toString()
```

Comment:
Looks like lightningcss used to type this as a `Buffer`, so `.toString()` here is correct. However it changed its type to `Uint8Array`, so using `.toString()` on it would be incorrect in the first place. Weirdly, lightningcss still returns a `Buffer` in Node.js.

Could we update this to use `TextDecoder` instead so we handle `Uint8Array` properly? That way we don't have to convert it to a `Buffer` and then convert it to a string. I also confirmed that `TextDecoder` works for `Buffer` too, but maybe good to test if it works with lightningcss altogether.

Reply:
I like the idea. I'll try to replace this line using `TextDecoder` and run tests. Let's see what happens :)

Reply:
**NodeJS: TextDecoder.decode vs Buffer.toString**

```ts
{
      "new TextDecoder().decode(res.code)": "h1 {\n" + "  font-size: 24px;\n" + "}\n" + "\n" + ".main {\n" + "  color: red;\n" + "  border: 1px solid gray;\n" + "}\n",
      "res.code.toString()": "h1 {\n" + "  font-size: 24px;\n" + "}\n" + "\n" + ".main {\n" + "  color: red;\n" + "  border: 1px solid gray;\n" + "}\n",
      "res.code": "<Buffer 68 31 20 7b 0a 20 20 66 6f 6e 74 2d 73 69 7a 65 3a 20 32 34 70 78 3b 0a 7d 0a 0a 2e 6d 61 69 6e 20 7b 0a 20 20 63 6f 6c 6f 72 3a 20 72 65 64 3b 0a 20 ... 27 more bytes>",
}
```

**Deno: TextDecoder.decode vs Uint8Array.toString**

```ts
{
      "new TextDecoder().decode(res.code)": "h1 {\n" + "  font-size: 24px;\n" + "}\n" + "\n" + ".gPMrEW_main {\n" + "  color: red;\n" + "  border: 1px solid gray;\n" + "}\n",
      "res.code.toString()": "104,49,32,123,10,32,32,102,111,110,116,45,115,105,122,101,58,32,50,52,112,120,59,10,125,10,10,46,103... 193 more characters",
      "res.code": Uint8Array(84)[(104, 49, 32, 123, 10, 32, 32, 102, 111, 110, 116, 45, 115, 105, 122, 101, 58, 32, 50, 52, 112, 120, 59, 10, 125, 10, 10, 46, 103, 80, 77, 114, 69, 87, 95, 109, 97, 105, 110, 32, 123, 10, 32, 32, 99, 111, 108, 111, 114, 58, 32, 114, 101, 100, 59, 10, 32, 32, 98, 111, 114, 100, 101, 114, 58, 32, 49, 112, 120, 32, 115, 111, 108, 105, 100, 32, 103, 114, 97, 121, 59, 10, 125, 10)],
}
```

Reply:
Thanks! That's a helpful comparison!